tests/test_transformers.py

```diff
@@ -0,0 +1,237 @@
+        assert_same_elements(
+            returned_elements,
+            example_edge_elements_from_client_map +
+            example_node_elements_from_client_map)
+
+
+def test_remove_highlighted_class_from_edges():
+    edge_map = {"source_name": [{"classes": "some_class"}]}
+    ujt.transformers.remove_highlighted_class_from_edges(edge_map)
+    assert edge_map["source_name"][0]["classes"] == ""
+
+
+def test_apply_highlighted_edge_class_to_edges():
+    source_names = ["source0", "source1", "source2", "source3"]
+    edge_map = defaultdict(
+        list,
+        {
+            source_names[source_idx]: [
+                {
+                    "data": {
+                        "target": source_names[source_idx + 1],
+                    },
+                    "classes": "",
+                }
+            ]
+            for source_idx in range(len(source_names) - 1)
+        })
+    ujt.transformers.apply_highlighted_class_to_edges(edge_map, source_names[0])
+    flattened_edges = [
+        edge for edge_list in edge_map.values() for edge in edge_list
+    ]
+    assert all(
+        ujt.constants.HIGHLIGHTED_UJ_EDGE_CLASS == edge["classes"]
+        for edge in flattened_edges)
+
+
+def test_apply_virtual_nodes_to_elements(patch_path, assert_same_elements):
+    # Node0 and Node1 are in collapsed VirtualNode0
+    # Node1 is a child of Node0
+    # Node2 in collapsed VirtualNode1
+    # Node0 has an edge to Node2
+    # Node1 has an edge to Node2
+    node_names = ["Node0", "Node1", "Node2"]
+    virtual_node_names = ["VirtualNode0", "VirtualNode1"]
+    node_elements = [
+        {
+            "data": {
+                "ujt_id": node_names[0],
+            },
+        },
+        {
+            "data": {
+                "ujt_id": node_names[1],
+                "parent": node_names[0],
+            },
+        },
+        {
+            "data": {
+                "ujt_id": node_names[2],
+            },
+        },
+    ]
+    edge_elements = [
+        {
+            "data": {
+                "source": node_names[0],
+                "target": node_names[2],
+            },
+        },
+        {
+            "data": {
+                "source": node_names[1],
+                "target": node_names[2],
+            },
+        },
+    ]
+    virtual_node_map = {
+        # apply_virtual_nodes_to_edges doesn't actually access the value inside
+        # the map, only the key. However, utils.get_highest_collapsed_virtual_node_name does.
+        virtual_node_names[0]:
+            VirtualNode(name=virtual_node_names[0],
+                        collapsed=True),
+        virtual_node_names[1]:
+            VirtualNode(name=virtual_node_names[1],
+                        collapsed=False),
+    }
+    parent_virtual_node_map = {
+        node_names[0]: virtual_node_names[0],
+        node_names[1]: virtual_node_names[0],
+        node_names[2]: virtual_node_names[1],
+    }
+
+    expected_elements = [
+        {
```

Comment:
not too sure about official python style but, would it make sense to have object generation helper functions here? I feel that perhaps having to define objects like this in each test might make the test less clear.

Reply:
Hm, I think it's better to explicitly define the expected output for tests that exercise transformer functions -- if we were to dynamically generate the expected output isn't that the same as implementing the same functionality as the code under test?

Reply:
you would still define the output explicitly, but instead of recreating a large object in each test to do so, you could maybe pass in whatever expected values to a constructor. alternatively, if we see similar objects being used in all test cases, perhaps a more global default object.

Reply:
dont think its a big issue, so leaving it up to you

Reply:
Sorry, misunderstood your point initially. 

Do you mean something like this?
```
def cyto_element_from_props(source, target, ujt_id)
    return { "source": source, ... }

expected_elements = [cyto_element_from_props(source1, target1, ujt_id1), ...]
```
It may be a little less verbose, but I think it's a little clearer to directly expose the structure of the dict object and makes the expected output more understandable

I do have some global input/output objects defined in [conftest.py](https://github.com/googleinterns/userjourneytool/pull/24/files#diff-484462fced51d1a06b1d93b4a44dd535) which are useful for when they need to be shared across multiple tests.

For small test cases like this, I prefer to define a set of inputs/outputs specific for the test for brevity and to isolate the properties being tested.